transifex/utils.py

```diff
@@ -297,7 +300,8 @@ def merge_pr(self):
                         'Failed to merge [%s/#%d], because of the exception [%s]',
                         self.name, self.pr.number, e,
                     )
-                    return False
+                    self.cleanup()
```

Comment:
The cleanup method is called in a finally block, so I don't think you need to call it explicitly here, or anywhere else.